src/components/general/NaiveSearchList/index.js

```diff
@@ -0,0 +1,160 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { _cs, isTruthyString } from '@togglecorp/fujs';
+
+import TextInput from '#rsci/TextInput';
+import FloatingContainer from '#rscv/FloatingContainer';
+import ListView from '#rscv/List/ListView';
+
+import {
+    calcFloatPositionInMainWindow,
+    defaultOffset,
+    defaultLimit,
+} from '#rsu/bounds';
+import _ts from '#ts';
+
+import styles from './styles.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    searchText: PropTypes.string,
+    onSearchChange: PropTypes.func,
+    list: PropTypes.array, // eslint-disable-line react/forbid-prop-types
+    listKeySelector: PropTypes.func,
+    listRenderer: PropTypes.func.isRequired,
+    listRendererParams: PropTypes.func.isRequired,
+    pending: PropTypes.bool,
+};
+
+const defaultProps = {
+    className: '',
+    searchText: '',
+    onSearchChange: () => {},
+    listKeySelector: d => d.id,
+    list: [],
+    pending: false,
+};
+
+export default class NaiveSearchList extends React.Component {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    constructor(props) {
+        super(props);
+
+        this.containerRef = React.createRef();
+        this.inputRef = React.createRef();
+        this.state = { showOptionsPopup: false };
+    }
+
+    componentDidMount() {
+        const { current: container } = this.containerRef;
+        if (container) {
+            this.boundingClientRect = container.getBoundingClientRect();
+        }
+    }
+
+    handleShowOptionsPopup = () => {
```

Comment:
Function name should be more clear

e.g. `handleTextInputFocus`, `handleTextInputClick`, `showOptionsPopup`, etc